staging/src/k8s.io/apiserver/pkg/server/options/serving.go

```diff
@@ -175,26 +171,6 @@ func (s *SecureServingOptions) ApplyTo(c *server.Config) error {
 		c.SecureServingInfo.SNICerts[server.LoopbackClientServerNameOverride] = &tlsCert
 	}
 
-	// create shared informers, if not explicitly set use in cluster config.
-	// do not fail on an error, this allows an external API server to startup
-	// outside of a kube cluster.
-	var clientCfg *rest.Config
```

Comment:
This appears to the crux of it here.  We had to add this block of ugly for an exceptional case, but now the exceptional case is leaking out to everyone.  Only the kube-apiserver (and federation if they're doing something a little crazy (they probably are)) needs to do this.  Maybe they don't get to use `ApplyTo`?